### PR TITLE
keystone: use os-client-config

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Comment by jsuchome** [comment]
If this works correctly, we should remove sourcing `.openrc` from upgrade bash scripts...

**Comment by rhafer** [comment]
> If this works correctly, we should remove sourcing .openrc from upgrade bash scripts

@jsuchome many (if not most) of the scripta run before the keystone recipes ran in their cloud7 variant. So unless we want to backport this to older crowbar releases. This won't really work.

**Comment by jsuchome** [comment]
Oh, true

**Comment by rhafer** [comment]
> the whole "don't forget to source ~/.openrc" is exceedingly annoying
> and obsolete since we have os-client-config since at least Liberty.

Don't you still need to source .openrc? Or at least set OS_CLOUD

**Comment by toabctl** [comment]
> Don't you still need to source .openrc? Or at least set OS_CLOUD

I think that is done now via /etc/profile.d . See https://github.com/crowbar/crowbar-openstack/pull/712/files#diff-28b6adf7bd1f81183dace3201a2720a5R486

**Comment by rhafer** [comment]
> I think that is done now via /etc/profile.d . See https://github.com/crowbar/crowbar-openstack/pull/712/files#diff-28b6adf7bd1f81183dace3201a2720a5R486

Oh, not sure how I overlooked that :/

**Comment by dirkmueller** [comment]
[(5) Update templates for config files for Newton - Tempest](https://trello.com/c/x99NxFiG/415-5-update-templates-for-config-files-for-newton-tempest)

**Comment by toabctl** [comment]
@dirkmueller failed while trying to boot the manila service instance:
```
12:37:23 ERROR (CommandError): You must provide a username or user ID via --os-username, --os-user-id, env[OS_USERNAME] or env[OS_USER_ID]
```

File: chef/cookbooks/keystone/recipes/server.rb [hunk 1]
```diff
@@ -464,3 +464,28 @@
     keystone_settings: KeystoneHelper.keystone_settings(node, @cookbook_name)
     )
 end
+
+directory "/root/.config/openstack/" do
+  owner "root"
+  group "root"
+  mode 0o755
```
Comment (vuntz) [inline chef/cookbooks/keystone/recipes/server.rb]: Any reason to not use 700 here?

Comment (dirkmueller) [inline chef/cookbooks/keystone/recipes/server.rb]: Yeah, can do

**Comment by vuntz** [comment]
@dirkmueller this needs a rebase

**Comment by dirkmueller** [comment]
(rebased after a while, lets give it another try for cloud 8)

File: chef/cookbooks/keystone/templates/default/crowbar-cloud.sh.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+[-f ~/.openrc ] && . ~/.openrc
```
Comment (cmurphy) [inline chef/cookbooks/keystone/templates/default/crowbar-cloud.sh.erb]: This needs a space after the [

File: chef/cookbooks/keystone/templates/default/keystone.conf.erb [hunk 3]
```diff
@@ -18,17 +18,32 @@ connection = <%= @sql_connection %>
...
+ca_certs = <%= @ssl_ca_certs %>
+=======
+wsgi_keep_alive = false
+
+[eventlet_server_ssl]
+enable = <%= @ssl_enable %>
+certfile = <%= @ssl_certfile %>
+keyfile = <%= @ssl_keyfile %>
+ca_certs = <%= @ssl_ca_certs %>
+>>>>>>> keystone: Cleanup spare empty lines in config template
```
Comment (cmurphy) [inline chef/cookbooks/keystone/templates/default/keystone.conf.erb]: rebase errors?

File: chef/cookbooks/keystone/templates/default/keystone.conf.erb [hunk 4]
```diff
@@ -18,17 +18,32 @@ connection = <%= @sql_connection %>
 idle_timeout = <%= @sql_idle_timeout %>
 
 [eventlet_server]
-public_bind_host = <%= @bind_service_host %>
-public_port = <%= @bind_service_port %>
-admin_bind_host = <%= @bind_admin_host %>
-admin_port = <%= @bind_admin_port %>
+public_workers = <%= [node["cpu"]["total"], 2, 4].sort[1] %>
+admin_workers = <%= [node["cpu"]["total"], 2, 4].sort[1] %>
```
Comment (cmurphy) [inline chef/cookbooks/keystone/templates/default/keystone.conf.erb]: seems unrelated?

File: chef/cookbooks/keystone/templates/default/openrc.erb [hunk 5]
```diff
@@ -1,16 +1,6 @@
-# OPENSTACK ENV VARIABLES
-export OS_USERNAME='<%= @keystone_settings['admin_user'] %>'
-export OS_PASSWORD='<%= @keystone_settings['admin_password'] %>'
+# This file is deprecated and not needed anymore
+# Credentials are now managed in ~/.config/openstack
+
+export OS_CLOUD='<%= node[:domain] %>-admin'
 export OS_ENDPOINT_TYPE='internalURL'
```
Comment (cmurphy) [inline chef/cookbooks/keystone/templates/default/openrc.erb]: This could be replaced with the "interface" property in clouds.yaml

**Comment by toabctl** [comment]
Closing. Please reopen if you still want to work on this.
